packages/p2p/src/components/modal-manager/modals/filter-modal/filter-modal.jsx

```diff
@@ -39,156 +89,317 @@ const FilterModal = () => {
+                                    large
+                                    secondary
+                                    onClick={() => {
+                                        buy_sell_store.onClickReset();
+                                        onClickClear();
+                                    }}
+                                >
+                                    {localize('Reset')}
+                                </Button>
+                                <Button
+                                    large
+                                    primary
+                                    onClick={() => {
+                                        buy_sell_store.onClickApply(selected_methods, selected_methods_text);
+                                        hideModal();
+                                    }}
+                                >
+                                    {localize('Apply')}
+                                </Button>
+                            </Button.Group>
+                        )}
+                    </Modal.Footer>
+                </Modal>
+            </DesktopWrapper>
+            <MobileWrapper>
+                <MobileFullPageModal
+                    body_className='filter-modal__body'
+                    is_modal_open={is_modal_open}
+                    height_offset='80px'
+                    is_flex
+                    header={<FilterModalHeader pageHeaderReturnFn={pageHeaderReturnFn} />}
+                    onClickClose={onClickClose}
+                    renderPageFooterChildren={() => {
+                        return (
+                            <React.Fragment>
+                                {buy_sell_store.show_filter_payment_methods ? (
+                                    <Button.Group className='filter-modal__footer-button-group'>
+                                        <Button
+                                            disabled={
+                                                !has_selected_payment_methods && !has_selected_all_payment_methods
+                                            }
+                                            large
+                                            secondary
+                                            onClick={onClickClear}
+                                        >
+                                            <Localize i18n_default_text='Clear' />
+                                        </Button>
+                                        <Button
+                                            disabled={!has_selected_payment_methods}
+                                            large
+                                            primary
+                                            onClick={() => {
+                                                buy_sell_store.onClickApply(selected_methods, selected_methods_text);
+                                                buy_sell_store.setShowFilterPaymentMethods(false);
```

Comment:
***suggestion***: could we move this into a function as I saw this was already done above as well